src/js/common/schemaform/save-in-progress/RoutedSavableReviewPage.jsx

```diff
@@ -46,6 +111,46 @@ class RoutedSavableReviewPage extends React.Component {
     }
   }
 
+  handleEdit = (pageKey, editing, index = null) => {
+    const fullPageKey = `${pageKey}${index === null ? '' : index}`;
+    if (editing) {
+      this.props.setViewedPages([fullPageKey]);
+    }
+    this.props.setEditMode(pageKey, editing, index);
+  }
+
+  handleSubmit = () => {
+    const formConfig = this.props.route.formConfig;
+    const { isValid, errors } = isValidForm(this.props.form, this.props.pagesByChapter);
+    if (isValid) {
+      this.props.submitForm(formConfig, this.props.form);
+    } else {
+      // validation errors in this situation are not visible, so we’d
+      // like to know if they’re common
+      if (this.props.form.data.privacyAgreementAccepted) {
+        recordEvent({
+          event: `${formConfig.trackingPrefix}-validation-failed`,
+        });
+        Raven.captureMessage('Validation issue not displayed', {
+          extra: {
+            errors,
+            prefix: formConfig.trackingPrefix
+          }
+        });
+        this.props.setSubmission('status', 'validationError');
+      }
+      this.props.setSubmission('hasAttemptedSubmit', true);
+    }
+  }
+
+  handleToggleChapter({ name, open, pageKeys }) {
+    if (open) {
+      this.props.closeReviewChapter(name, pageKeys);
+    } else {
+      this.props.openReviewChapter(name);
+    }
```

Comment:
was able to simplify this more by explicitly passing the `open` state on the props. Now this is just an `if` statement and should be easier to unit test.